drivers/video/ov7670.c

```diff
@@ -547,10 +550,73 @@ static int ov7670_init(const struct device *dev)
 	return 0;
 }
 
+static int ov7670_stream_start(const struct device *dev)
+{
+	return 0;
+}
+
+static int ov7670_stream_stop(const struct device *dev)
+{
+	return 0;
+}
+
+static int ov7670_set_ctrl_vflip(const struct device *dev, bool enable)
+{
+	int ret;
+	uint8_t reg = OV7670_MVFP;
+	const struct ov7670_config *config = dev->config;
+
+	ret = i2c_write_dt(&config->bus, &reg, sizeof(reg));
+	if (ret < 0) {
+		return ret;
+	}
+	ret = i2c_read_dt(&config->bus, &reg, sizeof(reg));
+	if (ret < 0) {
+		return ret;
+	}
+
+	reg = enable ? (reg | OV76770_MVFP_VFLIP) : (reg & ~OV76770_MVFP_VFLIP);
+	return i2c_reg_write_byte_dt(&config->bus, OV7670_MVFP, reg);
+}
+
+static int ov7670_set_ctrl_hmirror(const struct device *dev, bool enable)
+{
+	int ret;
+	uint8_t reg = OV7670_MVFP;
+	const struct ov7670_config *config = dev->config;
+
+	ret = i2c_write_dt(&config->bus, &reg, sizeof(reg));
+	if (ret < 0) {
+		return ret;
+	}
+	ret = i2c_read_dt(&config->bus, &reg, sizeof(reg));
+	if (ret < 0) {
+		return ret;
+	}
+
+	reg = enable ? (reg | OV76770_MVFP_MIRROR) : (reg & ~OV76770_MVFP_MIRROR);
+	return i2c_reg_write_byte_dt(&config->bus, OV7670_MVFP, reg);
+}
```

Comment:
That's a lot of duplicated code, can't there be a common helper function?

Reply:
Some sensors have a read-modify-write wrapper.
Similar review happened here, also in order to implement VFLIP/HFLIP:
- https://github.com/zephyrproject-rtos/zephyr/pull/81830#discussion_r1856545279

I hope this can be unified across all sensors in the long term. I will try something with IMX219 if time permits.

Reply:
Updated 👍🏼